doc/sonic-installer/SONiC_Image_Installation_Verification.md

```diff
@@ -0,0 +1,172 @@
+
+# SONiC Image Installation Verification
+
+# High Level Design Document
+
+#### Rev 0.1
+
+# Table of Contents
+
+- [Table of Contents](#table-of-contents)
+- [Revision](#revision)
+- [About this Manual](#about-this-manual)
+- [Scope](#scope)
+- [Definition/Abbreviation](#definition-abbreviation)
+    + [Table 1: Abbreviations](#table-1--abbreviations)
+- [1 Feature Overview](#1-feature-overview)
+- [2 Functional Requirements](#2-functional-requirements)
+- [3 Configuration and Management Requirements](#3-configuration-and-management-requirements)
+- [4 Design and Flow Diagrams](#4-design-and-flow-diagrams)
+  * [4.1 Flow Diagrams](#41-flow-diagrams)
+  * [4.2 Design](#42-design)
+    + [4.2.1 Generate platform list](#421-generate-platform-list)
+    + [4.2.2 Verification](#422-verification)
+    + [4.2.3 Skip platform check](#423-skip-platform-check)
+- [5 Error Handling](#5-error-handling)
+- [6 Serviceability and Debug](#6-serviceability-and-debug)
+- [7 Warm Boot Support](#7-warm-boot-support)
+- [8 Scalability](#8-scalability)
+- [9 Tests](#9-tests)
+  * [9.1 Unit Tests](#91-unit-tests)
+  * [9.2 End to End test](#92-end-to-end-test)
+
+
+# Revision
+
+| Rev | Date        | Author             | Change Description  |
+|:---:|:-----------:|:------------------:|---------------------|
+| 0.1  | 01/25/2021 | Jingwen Xie        | Initial version     |
+
+# About this Manual
+This document provides a detailed description on the strategy to implement the SONiC extended image installation verification feature.
+
+# Scope
+This document describes the high level design of a SONiC extended image installation verification feature. This document provides minor implementation details about the proposed solutions.
+
+# Definition/Abbreviation
+
+### Table 1: Abbreviations
+| **Term** | **Meaning**                                 |
+| -------- | ------------------------------------------- |
+| ASIC     | Application-Specific Integrated Circuit     |
+
+# 1 Feature Overview
+
+SONiC image installation requires users to install the correct image on the switch.  The image is built per ASIC type and it needs to be loaded on the corresponding switch of that ASIC type.
+
+For example, Dell6100’s switch ASIC vendor is Broadcom. It should be installed with a Broadcom based image. However, choosing  which image to install  on specific switch is simply determined by the users.  SONiC itself doesn’t have an extended verification step to verify if the image's built ASIC type is aligned with the switch ASIC type.
+
+Here is a list of ASIC vendor info from [Supported Devices and Platforms](https://github.com/Azure/SONiC/wiki/Supported-Devices-and-Platforms).
+
+The verification should looks like this:
+```
+admin@vlab-01:~$ sudo sonic-installer install sonic-broadcom.bin -y
+Image file 'sonic-broadcom.bin' is of a different platform ASIC type than running platform's.
+If you are sure you want to install this image, use --skip-platform-check.
+Aborting...
```

Comment:
>Aborting [](http://example.com/codeflow?start=0&length=8)

What is the diff of aborting and aborted? will it take long?

Reply:
It occurs at the same time. 
Aborting is from the log while after the log, it will raise abort.
```
echo_and_log("Image file '{}' is of a different platform ASIC type than running platform's.\n".format(url) +
                "If you are sure you want to install this image, use --skip-platform-check.\n" +
                "Aborting...", LOG_ERR)
            raise click.Abort()
```